### ISSUE TITLE
Thermia integration stop working after upgrade of HACS to 1.24.3

### ISSUE THREAD

**Issue by HenrikBurton** [klejejs/ha-thermia-heat-pump-integration#23]
Updated HACS to 1.24.3 and from there on there are no readings from my heatpump.

in the system CORE logg i get this message:

_2022-04-08 13:53:04 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 137, in _handle_refresh_interval
    await self._async_refresh(log_failures=True, scheduled=True)
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 268, in _async_refresh
    update_callback()
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 328, in _handle_coordinator_update
    self.async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 532, in async_write_ha_state
    self._async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 570, in _async_write_ha_state
    state = self._stringify_state(available)
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 538, in _stringify_state
    if (state := self.state) is None:
  File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 454, in state
    assert native_unit_of_measurement
AssertionError_

Any body else with the same issue?

**Comment by klejejs** [comment]
Hello! Did this break only after the HACS update or after the Home Assistant 2022.4.0 update? I assume it could be because the latest Home Assitant release changes how errors are perceived and as you had the error about getting temperature index, this might be now breaking the integration entirely. :( 

I have unfortunately been very busy for the past few weeks, but I'll definitely look into it this weekend, so it starts working for you once again.

**Comment by HenrikBurton** [comment]
You are probably right, I upgraded Home Assistant some hours earlier.

Appreciera your help 🙏

**Comment by jwictor81** [comment]
Got the same problem. No temperatures are shown.

Running
Versions
core-2022.4.1
supervisor-2022.03.5
Home Assistant OS 7.6

**Comment by pumpan** [comment]
Same here, after updating homeassistant.

And thx for a great integration!

**Comment by klejejs** [comment]
@HenrikBurton @jwictor81 @pumpan I just released an update that should fix the issue. Please check if temperatures are shown now.

**Comment by pumpan** [comment]
> @HenrikBurton @jwictor81 @pumpan I just released an update that should fix the issue. Please check if temperatures are shown now.

Yes its working again! Amazing that one change in 1 Line can do so much😆how did you find out that it was the problem? 

Again thx for a great integration!

**Comment by jwictor81** [comment]
> @HenrikBurton @jwictor81 @pumpan I just released an update that should fix the issue. Please check if temperatures are shown now.

Great work! Got them back once again.

Other question on the topic. I miss some sensors like return line. Is it known? BR

**Comment by pumpan** [comment]
> > @HenrikBurton @jwictor81 @pumpan I just released an update that should fix the issue. Please check if temperatures are shown now.
> 
> Great work! Got them back once again.
> 
> Other question on the topic. I miss some sensors like return line. Is it known? BR

Working for me.

**Comment by jwictor81** [comment]
Strange. Only got 11 entities where these 4 temperatures are available.

heat_target_temperature
hot_water_temperature
outdoor_temperature
supply_line_temperature

**Comment by HenrikBurton** [comment]
The fix seams to be working with all the sensors back and the error from issue #22 also disapered.

Great work @klejejs !

**Comment by klejejs** [comment]
Great to hear it's working!
@jwictor81 Let's discuss your sensors in another issue.
